Migrate base_import_async to 12.0

guewen opened the pull request:
* [x] needs #101

The first commits are shared with #101, relevant commits are the 2 last ones.

Commit:
Migrate base_import_async to 12.0

Commit:
Add setuptools files for base_import_async

lmignon commented on base_import_async/__manifest__.py:
```diff
@@ -22,5 +22,5 @@
         'static/src/xml/import.xml',
     ],
-    'installable': False,
+    'installable': True,
```
`development_status` ????

guewen replied:
added

lmignon reviewed (approved):
Thank you @guewen

Commit:
Add OCA development status

lmignon reviewed (commented):
Strange, the import functionality seems to not work at all on runbot....

guewen commented:
> Strange, the import functionality seems to not work at all on runbot....

Because of the CORS issue?

> An unknown CORS error occured. The error probably originates from a JavaScript file served from a > different origin. (Opening your browser console might give you a hint on the error.)

guewen commented:
On my tests on runbot, the jobs are generated but there is this CORS error that pops when the notification appears. And in the browser console I get:

```
uncaught exception: an exception occurred in a caller-provided callback function
XML Parsing Error: syntax error
Location: http://3341401-102-8f5137.runbot3.odoo-community.org/longpolling/poll
Line Number 1, Column 1:
```

There is a problem with the longpolling:
![selection_395](https://user-images.githubusercontent.com/417223/46604693-667cfc00-caf7-11e8-88de-9d5a6e0771b8.png)

The expected response for longpolling would have been something like (copied from odoo's runbot):
```
{"jsonrpc": "2.0", "id": 403105148, "result": [{"id": 406, "channel": ["381328-12-0-360212-all", "res.partner", 3], "message": {"type": "activity_updated", "activity_created": true}}]}
```

lmignon commented:
@guewen Yes the CORS issue is one problem but in my case I was not able to create the job...

lmignon commented:
@sbidoul @sbejaoui Do you use this addon on 12.0?

sbidoul commented:
@lmignon I've not tried it yet.

GSLabIt commented:
@guewen did u try to import products with url image? it seems to be failing, while importing "normally" it works properly.

<img width="1265" alt="Schermata 2019-09-18 alle 19 00 43" src="https://user-images.githubusercontent.com/11652845/65169509-f6852e80-da46-11e9-8d19-87ec45e85df3.png">

@lmignon 
> @guewen Yes the CORS issue is one problem but in my case I was not able to create the job...

it seems that, at least importing excel, `options` are not passed correctly; both `encoding` and `separator` are `''`, so it fails opening writer. 

Beside that i'm getting CORS problem as well ..

**edit:**: I can confirm issue problem _1-character string_ is raised by `encoding` and `separator` when importing excel file format. It will not forward `options` correctly; I fixed it having `encoding` failing to `'utf-8'` and `delimiter` to `','`.

GSLabIt commented:
@guewen any news on CORS issue?

guewen commented:
@GSLabIt No, I didn't have the opportunity to work on this.

Commit:
Exit the import screen after importing file in async

When calling super, the super method would fail with a 'state machine'
error, appearing confusingly as a CORS error client side. We do not
have to call super when we delayed import asynchronously, as anyway
the 'super' call would send an inaccurate confirmation notification and exit.

guewen commented:
I found the origin of the error (which is unrelated to CORS, only appearing as such), fixed in my last commit. I think we are ready to merge!

GSLabIt commented:
@guewen, great news. Did u try importing image from url from excel?

have a look at https://github.com/OCA/queue/pull/102#issuecomment-532776170

guewen commented:
> Did u try importing image from url from excel?

Nope, forgot about this one :) (never tried this actually)

guewen commented:
Actually, by looking at the implementation, I think that only CSV files are supported (which sort of make sense since files are split per line). I think we should hide the options and add a comment about this in the readme. (and if someone wants to implement excel file supports, it can be added later).

GSLabIt commented:
@guewen what about fixing it by having `encoding` failing to `'utf-8'` and `delimiter` to `','`?

Commit:
Set default values for encoding and separator

Import of Excel files do not set these options, although the keys
are in the dict.

Courtesy of @GSLabIt for finding the issue and solution.

guewen commented:
You are right, not sure how but excel files are read correctly. I pushed a new commit with this correction.

GSLabIt commented:
> You are right, not sure how but excel files are read correctly. I pushed a new commit with this correction.

That's because you'll have data parsed by: 
https://github.com/OCA/queue/blob/b6de8170c85aeb69040ae3eecc097c29052b9e85/base_import_async/models/base_import_import.py#L47

So first all file is parsed, than splitted. If  have very big file, parsing could end in timeout.

guewen commented:
Thanks @GSLabIt  :)

GSLabIt reviewed (approved):
LGTM

guewen commented:
/ocabot merge
